lib/api.js

```diff
@@ -68,20 +74,47 @@ function parseApiResponse(respData, callback) {
   callback(null, respData.result);
 }
 
+function retryApiRequest(args) {
+  if (!SETTINGS.retryInterval)
+    return;
+
+  retryQueue.push(args);
+
+  if (!retryHandle) {
+    retryHandle = setInterval(function() {
+      while (retryQueue.length) {
+        makeApiRequest(retryQueue.shift());
+      }
+    }, SETTINGS.retryInterval);
+  }
+}
+
+function dequeueApiRequest(req) {
+  for (var i=0; i < pendingQueue.length; i++) {
+    if (pendingQueue[i] == req) {
+      pendingQueue.splice(i, 1);
```

Comment:
This works because a request is only in the pendingQueue at most once, so if we find it (using ==) then we can splice it out of the array and return.